bin/genesis

```diff
@@ -1550,6 +1555,30 @@ sub detect_bosh_version {
 	}
 }
 
+sub system_bosh {
+	my ($cmd) = join(" ", ($BOSH, @_));
+	if ($ENV{BOSH_ALL_PROXY} ne '') {
+		$ENV{HTTPS_PROXY}='';
+	}
+	my @result = system_execute($cmd);
+	if ($ENV{BOSH_ALL_PROXY} ne '') {
+		$ENV{HTTPS_PROXY}=$ENV{BOSH_ALL_PROXY};
+	}
+	return @result;
+}
```

Comment:
This pattern of chage-then-restore is better handled using local scope, and the Perl `local` keyword.

If you declare a variable using `local` (as opposed to `my` or `our`), the binding will only persist until the stack is unwound at function return.

This function then becomes something like:

```
sub system_bosh {
  local $ENV{HTTP_PROXY} = ''; # bosh dislikes this env var
  return system_execute(join(" ", $BOSH, @_);
}
```

Reply:
Awesome. Much nicer.